rigs/yaesu/newcat.c

```diff
@@ -2193,491 +2413,244 @@ int newcat_get_rptr_shift(RIG *rig, vfo_t vfo, rptr_shift_t *rptr_shift)
-    }
-    else if (is_ftdx5000)
+    /* Search the lookup table for a matching entry */
+    for (i = 0; i < RPTR_OFFS_CMD_TABLE_SIZE; i++)
     {
-        if (freq >= 28000000 && freq <= 29700000)
-        {
-            strcpy(command, "EX081");
-        }
-        else if (freq >= 50000000 && freq <= 54000000)
-        {
-            strcpy(command, "EX082");
-        }
-        else
-        {
-            // only valid on 10m and 6m bands
-            RETURNFUNC(-RIG_EINVAL);
-        }
+        const rptr_offs_cmd_entry_t *entry = &rptr_offs_cmd_table[i];
 
-        // Step size is 1 kHz
-        offs /= 1000;
-
-        SNPRINTF(priv->cmd_str, sizeof(priv->cmd_str), "%s%04li%c", command, offs,
-                 cat_term);
-    }
-    else if (is_ftdx101d || is_ftdx101mp || is_ftdx10)
-    {
-        if (freq >= 28000000 && freq <= 29700000)
+        if (entry->rig_id != rig_id)
         {
-            strcpy(command, "EX010315");
-
-            if (is_ftdx10) { strcpy(command, "EX010317"); }
+            continue;  /* Not for this rig */
         }
-        else if (freq >= 50000000 && freq <= 54000000)
-        {
-            strcpy(command, "EX010316");
 
-            if (is_ftdx10) { strcpy(command, "EX010318"); }
-        }
-        else
+        /*
+         * Check if this entry matches the band:
+         * - RPTR_BAND_ALL_HF matches any frequency (for FT-450 which uses
+         *   a single command for all bands)
+         * - Otherwise, entry->band must match the frequency's band
+         */
+        if (entry->band == RPTR_BAND_ALL_HF || entry->band == band)
         {
-            // only valid on 10m and 6m bands
-            RETURNFUNC(-RIG_EINVAL);
+            /* Found a matching entry */
+            SNPRINTF(cmd, RPTR_CMD_MAXLEN, "%s", entry->cmd);
+            *step = entry->step;
+            return RIG_OK;
         }
-
-        // Step size is 1 kHz
-        offs /= 1000;
-
-        SNPRINTF(priv->cmd_str, sizeof(priv->cmd_str), "%s%04li%c", command, offs,
-                 cat_term);
     }
-    else
+
+    /*
+     * No matching entry found. This could mean:
+     * 1. The rig doesn't support repeater offset commands (return -RIG_ENAVAIL)
+     * 2. The frequency is not in a valid band for this rig (return -RIG_EINVAL)
+     *
+     * Check if the rig has ANY entries in the table to distinguish these cases.
+     */
+    for (i = 0; i < RPTR_OFFS_CMD_TABLE_SIZE; i++)
     {
-        RETURNFUNC(-RIG_ENAVAIL);
+        if (rptr_offs_cmd_table[i].rig_id == rig_id)
+        {
+            /* Rig is in the table but frequency is not in a valid band */
+            return -RIG_EINVAL;
+        }
     }
 
-    RETURNFUNC(newcat_set_cmd(rig));
+    /* Rig is not in the table at all */
+    return -RIG_ENAVAIL;
 }
 
 
-int newcat_get_rptr_offs(RIG *rig, vfo_t vfo, shortfreq_t *offs)
+/*
+ * =============================================================================
+ * REFACTORED: newcat_set_rptr_offs (2025)
```

Comment:
This comment section can be removed, it doesn't really document the function at all.